### ISSUE TITLE
"Generate variant" inserts code into current file, not file with enum definition

### ISSUE THREAD

**Issue by TimNN** [rust-lang/rust-analyzer#12372]
**rust-analyzer version**: `rust-analyzer version: 84be2eaf9 2022-05-23 stable`

When using the "Generate [enum] variant" assist, the new enum variant is inserted into the current file.

It should instead be inserted into the file that defines the enum (which may not be the current file).

cc @fasterthanlime

**Comment by fasterthanlime** [comment]
Yup that's certainly a failure mode neither the original author nor me not the reviewer thought about 😌

I might take it up if nobody else does — a pointer to an assist that does it correctly would probably be a time saver!

**Comment by Veykril** [comment]
Whoops. Fixing that is simple, the `adt.source(ctx.db())?.original_ast_node(ctx.db())?` in
https://github.com/rust-lang/rust-analyzer/blob/d7c147406eff20783abba6ff1fe250ecc44cf800/crates/ide-assists/src/handlers/generate_enum_variant.rs#L66
returns an `InFile` which contains a file id, then it's just a matter of calling `builder.edit_file` with that before doing the edits.
